curriculum/challenges/english/25-front-end-development/lab-bookmark-manager-app/66def5467aee701733aaf8cc.md

```diff
@@ -0,0 +1,781 @@
+} finally {
+  resetLocalStorage();
+}
+```
+
+Each `label` element should contain an anchor element with the bookmark name as text, and the `href` attribute set to the bookmark URL.
+
+```js
+setLocalStorage();
+try {
+  categoryDropdownTest.value = "news";  
+  viewCategoryButtonTest.dispatchEvent(new Event("click"));
+  const bookmarkAnchorsDisplayed = document.querySelectorAll('#category-list label>a');
+  assert.lengthOf(bookmarkAnchorsDisplayed, 2);
+  assert.strictEqual(bookmarkAnchorsDisplayed[0].innerText, "example1");
+  assert(bookmarkAnchorsDisplayed[0].href.endsWith("example1.com"));
+  assert.strictEqual(bookmarkAnchorsDisplayed[1].innerText, "example4");
+  assert(bookmarkAnchorsDisplayed[1].href.endsWith("example4.com"));
+} finally {
+  resetLocalStorage();
+}
+```
+
+When you click `#close-list-button`, you should hide `#bookmark-list-section` and display the main section.
+
+```js
+viewCategoryButtonTest.dispatchEvent(new Event("click"));
+assert.strictEqual(getHidden("bookmark list"), "main");
+closeListButtonTest.dispatchEvent(new Event("click"));
+assert.strictEqual(getHidden("bookmark list"), "bookmark list");
+```
+
+When you click `#delete-bookmark-button`, you should delete the bookmark corresponding to the selected radio button from the local storage and update the displayed bookmark list.
+
+```js
+setLocalStorage();
+try {
+  categoryDropdownTest.value = "news";  
+  viewCategoryButtonTest.dispatchEvent(new Event("click"));
+  const firstRadioButton = document.querySelector('#category-list input[type="radio"]');
+  firstRadioButton.checked = true;
+  deleteBookmarkButton.dispatchEvent(new Event("click"));
+  let bookmarksDisplayed = document.querySelectorAll('#category-list input[type="radio"]');
+  let bookmarkLabelsDisplayed = document.querySelectorAll('#category-list label');
+  assert.lengthOf(bookmarksDisplayed, 1);
+  assert.strictEqual(bookmarksDisplayed[0].id, "example4");
+  assert.strictEqual(bookmarksDisplayed[0].value, "example4");
+  assert.lengthOf(bookmarkLabelsDisplayed, 1);
+  assert.strictEqual(bookmarkLabelsDisplayed[0].htmlFor, "example4");
+  const expected = [{name: "example2", category: "entertainment", url: "example2.com"}, {name: "example3", category: "work", url: "example3.com"}, {name: "example4", category: "news", url: "example4.com"}];
+  assert.deepEqual(getBookmarks(), expected);
+  
+} finally {
+  resetLocalStorage();
+}
+```
+
+# --seed--
+
+## --after-user-code--
+
+```js
+const viewCategoryButtonTest = document.getElementById("view-category-button");
+const closeListButtonTest = document.getElementById("close-list-button");
+const closeFormButtonTest = document.getElementById("close-form-button");
+const addBookMarkButtonTest = document.getElementById("add-bookmark-button");
+const addBookmarkButtonFormTest = document.getElementById("add-bookmark-button-form");
+const deleteBookmarkButtonTest = document.getElementById("delete-bookmark-button");
```

Comment:
```suggestion
const deleteBookmarkButton = document.getElementById("delete-bookmark-button");
```

Not sure if this is the actual issue, but the last test was failing with the error: 

```js
// console output
[ReferenceError: deleteBookmarkButton is not defined]
```